apps/etterlatte-samordning-vedtak/src/main/kotlin/no/nav/etterlatte/samordning/vedtak/LoggingPlugins.kt

```diff
@@ -0,0 +1,113 @@
+package no.nav.etterlatte.samordning.vedtak
+
+import io.ktor.server.application.ApplicationCall
+import io.ktor.server.application.ApplicationCallPipeline
+import io.ktor.server.application.Hook
+import io.ktor.server.application.RouteScopedPlugin
+import io.ktor.server.application.call
+import io.ktor.server.application.createApplicationPlugin
+import io.ktor.server.application.createRouteScopedPlugin
+import io.ktor.server.application.hooks.ResponseSent
+import io.ktor.server.request.httpMethod
+import io.ktor.server.request.path
+import io.ktor.server.request.uri
+import io.ktor.util.AttributeKey
+import io.ktor.util.logging.KtorSimpleLogger
+import io.ktor.util.pipeline.PipelinePhase
+import net.logstash.logback.marker.Markers
+import no.nav.etterlatte.libs.ktor.PluginConfiguration
+import no.nav.etterlatte.libs.ktor.brukerTokenInfo
+import no.nav.etterlatte.token.Saksbehandler
+import no.nav.etterlatte.token.Systembruker
+import org.slf4j.MDC
+
+internal val LOGGER = KtorSimpleLogger("no.nav.etterlatte.samordning.requestLogger")
+
+private val userAttribute = AttributeKey<String>("user")
+private val startTimeAttribute = AttributeKey<Long>("starttime")
+private val loggingPerformed = AttributeKey<Boolean>("requestLoggingPerformed")
+
+private object UserIdMdcHook : Hook<suspend (ApplicationCall) -> Unit> {
+    private val UserIdMdcHook: PipelinePhase = PipelinePhase("UserIdMdc")
+    private val AuthenticatePhase: PipelinePhase = PipelinePhase("Authenticate")
+
+    override fun install(
+        pipeline: ApplicationCallPipeline,
+        handler: suspend (ApplicationCall) -> Unit,
+    ) {
+        // Se Tilgangsstyring.kt
+        pipeline.insertPhaseAfter(ApplicationCallPipeline.Plugins, AuthenticatePhase)
+        pipeline.insertPhaseAfter(AuthenticatePhase, UserIdMdcHook)
+        pipeline.intercept(UserIdMdcHook) { handler(call) }
+    }
+}
+
+val userIdMdcPlugin: RouteScopedPlugin<PluginConfiguration> =
+    createRouteScopedPlugin(
+        name = "UserIdMdcPlugin",
+        createConfiguration = ::PluginConfiguration,
+    ) {
+        on(UserIdMdcHook) { call ->
+            val user =
+                if (call.request.uri.contains("pensjon")) {
+                    when (val bruker = call.brukerTokenInfo) {
+                        is Systembruker -> bruker.sub
+                        is Saksbehandler -> bruker.ident
+                        else -> "ukjent"
+                    }
+                } else {
+                    call.orgNummer
+                }
+
+            MDC.put("user", user)
+            call.attributes.put(userAttribute, user)
+
+            return@on
+        }
+    }
+
+val serverRequestLoggerPlugin =
```

Comment:
Det kjens som heile denne er noko som burde fins i ktor alt, og kunne aktiverast med eit eller anna konfig-flagg, men kan godt vera eg berre er for godt vant frå andre applikasjonsserverar.

Reply:
Det er bare det at det er ikke godt nok for hva man trenger å få inn i kibana